### PR TITLE
Build workflow

### PR DESCRIPTION
**Pull request by celery6**
hopefully last repo setup for now...

### CODE REVIEW

File: .cproject
Comment (martendo) [inline .cproject:1]: I know nothing about this file but from a quick glance it seems like there are still paths that are not workspace-relative?
https://github.com/waterloo-rocketry/cansw_processor_canards/blob/24e393621ed5dc9af0ae270788f6746b96e086b2/.cproject#L170-L182

File: .cproject [hunk 1]
```diff
@@ -59,39 +59,24 @@
...
+									<listOptionValue builtIn="false" value="&quot;${workspace_loc:/processor_board/Drivers/CMSIS/Include}&quot;"/>
+									<listOptionValue builtIn="false" value="&quot;${workspace_loc:/processor_board/Drivers/CMSIS/Device/ST/STM32H7xx/Include}&quot;"/>
+									<listOptionValue builtIn="false" value="&quot;${workspace_loc:/processor_board/FATFS/Target}&quot;"/>
+									<listOptionValue builtIn="false" value="&quot;${workspace_loc:/processor_board/FATFS/App}&quot;"/>
+									<listOptionValue builtIn="false" value="&quot;${workspace_loc:/processor_board/Middlewares/Third_Party/FreeRTOS/Source/include}&quot;"/>
+									<listOptionValue builtIn="false" value="&quot;${workspace_loc:/processor_board/Middlewares/Third_Party/FreeRTOS/Source/CMSIS_RTOS_V2}&quot;"/>
+									<listOptionValue builtIn="false" value="&quot;${workspace_loc:/processor_board/Middlewares/Third_Party/FreeRTOS/Source/portable/GCC/ARM_CM4F}&quot;"/>
+									<listOptionValue builtIn="false" value="&quot;${workspace_loc:/processor_board/Middlewares/Third_Party/FatFs/src}&quot;"/>
+									<listOptionValue builtIn="false" value="&quot;${workspace_loc:/processor_board/Core/ThreadSafe}&quot;"/>
```
Comment (martendo) [inline .cproject:78]: Is there a difference between using `${ProjName}` and `processor_board`? Is this autogenerated and it doesn't even matter?

File: .cproject
Comment (celery6) [reply .cproject:1]: those ones are autogenerated, i think ill leave them.

File: .cproject (hunk 1 above)
Comment (celery6) [reply .cproject]: autogenerated - will probably get overwritten next time we regen from ioc

**Review by martendo — APPROVED** [review]

**Review by JasonBrave — APPROVED** [review]
LGTM :+1:

**Review by Redstone-ray — APPROVED** [review]

**Pull request merged**